File: Typescript/groupAnagrams.ts
function groupAnagrams(strs: string[]): string[][] {
    const map = new Map<string, string[]>();

    for (const str of strs) {
        const sortedStr = str.split('').sort().join('');
        if (!map.has(sortedStr)) {
            map.set(sortedStr, []);
        }
        map.get(sortedStr)!.push(str);
    }

    const result: string[][] = [];
    for (const group of map.values()) {
        result.push(group);
    }

    return result;
}

File: C++/groupAnagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        std::unordered_map<std::string, std::vector<std::string>> mp;

        for (auto str : strs) {
            auto strCpy = str;
            std::sort(strCpy.begin(), strCpy.end());
            mp[strCpy].push_back(str);
        }

        std::vector<std::vector<std::string>> res;
        for(auto pair : mp) {
            res.push_back(pair.second);
        }
        return res;
    }
};